tool/tsh/db.go

```diff
@@ -41,19 +42,63 @@ func onListDatabases(cf *CLIConf) error {
 	if err != nil {
 		return trace.Wrap(err)
 	}
+
 	var databases []types.Database
 	err = client.RetryWithRelogin(cf.Context, tc, func() error {
 		databases, err = tc.ListDatabases(cf.Context)
 		return trace.Wrap(err)
 	})
+
 	if err != nil {
 		return trace.Wrap(err)
 	}
+
 	// Retrieve profile to be able to show which databases user is logged into.
 	profile, err := client.StatusCurrent(cf.HomePath, cf.Proxy)
 	if err != nil {
 		return trace.Wrap(err)
 	}
+
+	// calculating possible users for db is only done in verbose mode.
+	displayUsersForDb := func(database types.Database) string {
+		return ""
+	}
+
+	if cf.Verbose {
+		// in case of any errors down the line, indicate that fact.
+		displayUsersForDb = func(database types.Database) string {
+			return "(error: unable to check)"
+		}
+
+		err = client.RetryWithRelogin(cf.Context, tc, func() error {
+			proxy, err := tc.ConnectToProxy(cf.Context)
+			if err != nil {
+				return trace.Wrap(err)
+			}
```

Comment:
nit: move to a separate function also the logic can be splitted fetching roles, making a  `displayUsersForDb` function.